File: src/plugin/combat/magic/lunar/teleport/MoonclanTeleportSpell.java
package plugin.combat.magic.lunar.teleport;

import org.redrune.game.global.WorldTile;
import org.redrune.game.plugin.combat.spell.type.TeleportSpellPlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 7/27/2017
 */
public class MoonclanTeleportSpell extends TeleportSpellPlugin {
	
	@Override
	public WorldTile destination() {
		return new WorldTile(2100, 3915, 0);
	}
	
	@Override
	public int levelRequired() {
		return 69;
	}
	
	@Override
	public int[] runesRequired() {
		return arguments(ASTRAL_RUNE, 2, LAW_RUNE, 1, EARTH_RUNE, 2);
	}
	
	@Override
	public int spellId() {
		return 43;
	}
	
	@Override
	public double exp() {
		return 66;
	}
	
	@Override
	public MagicBook book() {
		return MagicBook.LUNAR;
	}
}

File: src/main/java/plugin/interaction/combat/magic/lunar/teleport/MoonclanTeleportSpell.kt
package plugin.interaction.combat.magic.lunar.teleport

import org.redrune.game.content.plugin.combat.spell.type.TeleportSpellPlugin
import org.redrune.game.global.WorldTile
import org.redrune.utility.constants.MagicConstants
import org.redrune.utility.constants.MagicConstants.MagicBook

/**
 * @author Tyluur <itstyluur></itstyluur>@icloud.com>
 * @since 7/27/2017
 */
class MoonclanTeleportSpell : TeleportSpellPlugin {
    override fun destination(): WorldTile {
        return WorldTile(2100, 3915, 0)
    }

    override fun levelRequired(): Int {
        return 69
    }

    override fun runesRequired(): IntArray {
        return arguments(MagicConstants.ASTRAL_RUNE, 2, MagicConstants.LAW_RUNE, 1, MagicConstants.EARTH_RUNE, 2)
    }

    override fun spellId(): Int {
        return 43
    }

    override fun exp(): Double {
        return 66.0
    }

    override fun book(): MagicBook {
        return MagicBook.LUNAR
    }
}